extensions/migrate.md

```diff
@@ -0,0 +1,321 @@
+acknowledging the request: the server simply starts a migration if the request
+is accepted.
+
+## START
+
+The server sends a `MIGRATE START` message to the client to indicate that a
+migration is starting.
+
+    :saguaro.desert MIGRATE START cactus.desert-network.com +6697 5AEAD4D0
+    9DD9F7F5 :Server going down for maintenance
+
+The parameters are
+
+1. `START`, identifying the command
+
+2. The client-reachable address of the server that the client will eventually be
+   migrated to. This can be either a domain name or an IP address, but should be
+   reachable by the client.
+
+3. The port that the client should use when connecting to the new server. An
+   ASCII plus character (`+`) before the port indicates that TLS should be used.
+
+4. The *resume token*, to be used by the client when connecting to the new
+   server.
+
+5. The *confirm token* token, to be presented by the new server upon migration.
+
+6. An optional message from the server indicating the reason for the migration.
+
+Clients should respond with `MIGRATE OK` as soon as possible. Until `MIGRATE OK`
+has been received, the server should continue processing messages from the
+client normally.
+
+## OK
+
+The client sends `MIGRATE OK` to indicate it's prepared to start the migration
+process.
+
+    MIGRATE OK 5AEAD4D0
+
+The parameters are
+
+1. `OK`, identifying the operation
+
+2. The server-assigned resume token, given by the corresponding `MIGRATE START`.
+
+`MIGRATE OK` is the last message sent by the client to the old server. Upon
+receiving this message, the old server is clear to begin the process of
+migrating the client to the new server. However, the client should not attempt
+to resume their session until receiving a `MIGRATE PROCEED` message.
+
+## PROCEED
+
+The old server sends `MIGRATE PROCEED` to indicate that the client is cleared
+to connect to the new server and resume the session.
+
+    :saguaro.desert MIGRATE PROCEED 5AEAD4D0
+
+The parameters are
+
+1. `PROCEED`, identifying the operation.
+
+2. The server-assigned resume token, sent in the initial `MIGRATE START`
+   message.
+
+`MIGRATE PROCEED` is the last message sent by the old server. The old connection
+is now useless and both the client and server are free to close the connection
+at their leisure.
+
+## RESUME
+
+Client send `MIGRATE RESUME` to new servers to pick up a migrated session.
+
+    MIGRATE RESUME 5AEAD4D0
+
+The parameters are
+
+1. `RESUME`, identifying the operation.
+
+2. The resume token that was assigned to this migration in the initial
+   `MIGRATE START` message.
+
+This message should be sent as the first message when opening a connection with
+the new server, in place of normal registration messages like `NICK` and `CAP`.
+
+To ensure that the new server is prepared to accept the migration, clients
+should also send normal registration commands after `MIGRATE RESUME`:
+
+    MIGRATE RESUME 5AEAD4D0
+    CAP LS
+    NICK aji
+    USER alex * * :Alex
```

Comment:
This contradicts what you wrote above: 

> 4. On connecting to the new server, instead of using the normal connection setup
>   protocol (`NICK`, `CAP`, etc.) the client sends a `MIGRATE RESUME` message